libV2/redis.js

```diff
@@ -15,6 +15,49 @@ const moduleLogger = new LoggerContext({
 const COMMAND_TIMEOUT = 10000;
 const CONNECTION_TIMEOUT = 30000;
 
+/**
+ * Promisifies a function. If the function already returns a promise,
+ * it returns it as is. Handles both callbacks and promises.
+ *
+ * @param {Function} originalFn The function to promisify.
+ * @returns {Function} A function that returns a promise.
+ */
+function flexiblePromisify(originalFn) {
```

Comment:
Is this a standard, proven function?
- if so, can you provide the ref
- if not, can you add unit tests to make sure it works as designed?